File: algorithms/bubble_sort/bubble_sort.go
package main

import "fmt"

func BubbleSort(inputArr []int) []int {
	unsortedUntil := len(inputArr) - 1
	sorted := false

	for sorted != true {
		sorted = true
		for i := 0; i < unsortedUntil; i++ {
			if inputArr[i] > inputArr[i+1] {
				inputArr[i], inputArr[i+1] = inputArr[i+1], inputArr[i]
				sorted = false
			}
		}
		unsortedUntil--
	}
	return inputArr
}

func main() {
	fmt.Println(BubbleSort([]int{65, 55, 45, 35, 25, 15, 10}))
	fmt.Println(BubbleSort([]int{}))
}

File: algorithms/bubble_sort/bubble_sort.py
def bubble_sort(arr):
    bound = len(arr) - 1
    sorted = False
    
    while not sorted:
        sorted = True
        for i in range(bound):
            if arr[i] > arr[i+1]:
                arr[i], arr[i+1] = arr[i+1], arr[i]
                # If we do one modification, this means the array is not sorted
                sorted = False
        # Reduce the bound one step
        bound -= 1
    
    return arr

print(bubble_sort([65, 55, 45, 35, 25, 15, 10]))
